examples/connecting/sasl_ssl_plain.go

```diff
@@ -0,0 +1,50 @@
+package connect
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/twmb/kafka-go/pkg/kgo"
+	"github.com/twmb/kafka-go/pkg/sasl/plain"
+)
+
+func connectSaslSslPlain() {
+	fmt.Println("starting...")
+
+	seeds := []string{"localhost:9092"}
+	// SASL Plain credentials
+	user := ""
+	password := ""
+
+	opts := []kgo.Opt{
+		kgo.SeedBrokers(seeds...),
+
+		// SASL Options
+		kgo.SASL(plain.Plain(func(context.Context) (plain.Auth, error) {
+			return plain.Auth{
+				User: user,
+				Pass: password,
+			}, nil
+		})),
+
+		// Configure TLS. Uses SystemCertPool for RootCAs by default.
+		dialer := &tls.Dialer{
```

Comment:
This dialer needs to be above the `kgo.Opt[]`, but alternatively, you could to
```go
kgo.Dialer((&tls.Dialer{
  NetDialer: &net.Dialer{Timeout: 10*time.Second},
}).DialContext)
```

If you instead move the dialer above `kgo.Opt`, this can be
```go
kgo.Dialer(dialer.DialContext)
```